packages/expo-insights/ios/InsightsModule.swift

```diff
@@ -0,0 +1,107 @@
+import ExpoModulesCore
+import EASClient
+
+/**
+ Indicates whether the app launch event has already been sent.
+ */
+private var wasAppLaunchEventDispatched = false
+
+public final class InsightsModule: Module {
+  public func definition() -> ModuleDefinition {
+    Name("ExpoInsights")
+
+    OnCreate {
+      // The app launch event should be sent only during the first launch
+      // which means that we need to prevent dispatching them on app reload.
+      if !wasAppLaunchEventDispatched {
+        wasAppLaunchEventDispatched = true
+
+        Task {
+          try await dispatchLaunchEvent()
+        }
+      }
+    }
+  }
+
+  /**
+   Sends the `APP_LAUNCH` event.
+   */
+  private func dispatchLaunchEvent() async throws {
+    guard let manifest = appContext?.constants?.constants()["manifest"] as? [String: Any] else {
```

Comment:
this gets the build-time manifest which is in the classic format still (I think). Either way using the `expo-manifests` package should help here.

https://linear.app/expo/issue/ENG-8510/address-post-commit-comments-from-httpsgithubcomexpoexpopull21932